core/src/main/scala/dagr/core/tasksystem/Schedulable.scala

```diff
@@ -48,6 +48,65 @@ trait Schedulable {
+      }.headOption
+    }
+  }
+
+  /** A crude attempt at estimating the minimum memory needed by assuming a fixed maximum number of cores. */
+  private[tasksystem] def minMemory(start: Memory = Memory("256m"), end: Memory = Memory("64g"), step: Memory = Memory("256m"), maxCores: Cores = Cores(32)): Option[Memory] = {
+    // first check that the search will succeed, then test a range of values for memory
+    this.pickResources(availableResources = new ResourceSet(maxCores, end)).flatMap { _ =>
+      (start.value to end.value by step.value).flatMap { memory =>
+        val availableResources = new ResourceSet(maxCores, Memory(memory))
+        this.pickResources(availableResources = availableResources).map(_.memory)
+      }.headOption
+    }
+  }
+
+  /** A crude attempt at estimating the minimum resources by assuming a fixed ratio of memory per core. */
+  private[tasksystem] def minCoresAndMemory(start: Cores = Cores(1), end: Cores = Cores(32), step: Cores = Cores(1), memoryPerCore: Memory = Memory("1g")): Option[ResourceSet] = {
+    (start.value to end.value by step.value).flatMap { cores =>
+      val availableResources = new ResourceSet(Cores(cores), Memory((memoryPerCore.value * cores).toLong))
+      this.pickResources(availableResources = availableResources)
+    }.headOption
+  }
+
+  /**
+    * Given a non-null ResourceSet representing the maximum available resources return either a
+    * ResourceSet that is a subset of the available resources in which the task can run, or None
+    * if the task cannot run in the maximum resources.
+    *
+    * First estimates the # of cores by assuming an infinite amount of memory, and then estimates
+    * the amount of memory assuming an infinite # of cores.  If either does not yield a value or if
+    * the task cannot run with the combination of the two, then try a various combinations of cores
+    * and memory where we assume a fixed ratio between cores and memory (ex. 1g per core).
+    *
+    * @param maximumResources The maximum system resources available to the task.
+    * @return Either a ResourceSet of the subset of resources that this task can run with, or None
+    */
+  private[core] def minResources(maximumResources: ResourceSet = new ResourceSet(cores=Cores(32), memory=Memory("64g"))): Option[ResourceSet] = {
+    val cores  = minCores(end=maximumResources.cores, maxMemory=maximumResources.memory)
+    val memory = minMemory(end=maximumResources.memory, maxCores=maximumResources.cores)
+    (cores, memory) match {
+      case (Some(c), Some(m)) => this.pickResources(maximumResources.copy(cores=c, memory=m))
+      case _                  =>
+        // try a combination of cores and memory with a fixed reatio between the two.
+        val startMemoryPerCore = Memory("256m").value
+        val endMemoryPerCore   = maximumResources.memory.value
+        val stepMemoryPerCore  = Memory("256m").value
```

Comment:
Instead of hard-coding this to 256m, could you set it to be some reasonable fraction (e.g. 1/100th) of the max memory so you have a fixed number of steps?